### DISCUSSION TITLE
File picker on web

### DISCUSSION THREAD

**Discussion by capidreamliner** [flet-dev/flet#2412] [Q&A]
### Question

Hey there!

I'm new to Flet, and I'm having a bit of trouble with something. Sorry if it's a simple question, but I have searched everywhere and can't find a solution. I'm working on an app using Flet and Pyrebase, and I want users to be able to pick an image and put it on Firebase Storage.

Everything's works great except when I run it on the web. In my code, I use file.path to upload the file, but on the web, it just returns None because of security reasons. How can I upload the file without using the path?

Sorry if it's a bother, and I appreciate any help in advance!

### Code sample

```python
def on_file_picker_result(e: ft.FilePickerResultEvent):
        if e.files:
            file = e.files[0]
            storage.child("files/as").put(file.path)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Web upload is explained here: https://flet.dev/docs/guides/python/file-picker-and-uploads#uploading-files
On that same page, the reason you got None is mentioned:
> When running Flet app in a browser only "Pick files" option is available and it's used for uploads only as it, obviously, doesn't return a full path to a selected file.

**Reply by capidreamliner** [reply]
I appreciate your response.

I read that documentation before asking my question, and there is no clear mention of how to handle this type of uploads. I can't use FilePickerUploadFile because I'm using Pyrebase and I need to pass the user authentication token.

Thank you very much!

**Reply by bobwatcherx** [reply]
why this error . and how to get path file for upload @ndonkoHenri 

code
```
from flet import *

def main(page:Page):
    def upload_files(e):
        upload_list = []
        if file_picker.result != None and file_picker.result.files != None:
            for f in file_picker.result.files:
                upload_list.append(
                    FilePickerUploadFile(
                        f.name,
                        upload_url=page.get_upload_url(f.name, 600),
                    )
                )
            file_picker.upload(upload_list)

    page.add(
    ElevatedButton("Upload", on_click=upload_files)

    )
app(main,view=AppView.WEB_BROWSER,port=5050)
```

```
Traceback (most recent call last):
  File "C:\Users\LENOVO\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "C:\Users\LENOVO\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\LENOVO\Pictures\fletapp\webfletupload\main.py", line 6, in upload_files
    if file_picker.result != None and file_picker.result.files != None:
       ^^^^^^^^^^^^^^^^^^
AttributeError: module 'flet_core.file_picker' has no attribute 'result'
```

**Reply by ndonkoHenri** [reply]
Seems like file_picker is neither defined in your code, nor was added to the page's overlay. 
Try the example in the docs.
